pkg/servicescm/servicescm.go

```diff
@@ -158,14 +166,21 @@ func Generate(name, namespace string, data *Data) (*core.ConfigMap, error) {
 	}
 	servicesConfigMap.Data[filesKey] = string(jsonFiles)
 
+	if data.EnvironmentVars != nil {
+		jsonEnvVars, err := json.Marshal(data.EnvironmentVars)
+		if err != nil {
+			return nil, err
+		}
+		servicesConfigMap.Data[envVarsKey] = string(jsonEnvVars)
+	}
 	return servicesConfigMap, nil
 }
 
 // Parse converts ConfigMap data into the objects representing a Windows services ConfigMap schema
 // Returns error if the given data is invalid in structure
 func Parse(dataFromCM map[string]string) (*Data, error) {
-	if len(dataFromCM) != 2 {
-		return nil, fmt.Errorf("services ConfigMap should have exactly 2 keys")
+	if len(dataFromCM) != 2 && len(dataFromCM) != 3 {
+		return nil, fmt.Errorf("services ConfigMap can only have 2 or 3 number of keys")
```

Comment:
Please add a comment explaining this.
Is this to provide compatibility when upgrading from a ConfigMap from a previous version? We need to make sure thats captured and removed when necessary. Making this less restrictive should also be considered.

Reply:
The environmentVars key is optional in the services CM (present when a proxy is enabled). I'll add a comment explaining this